workers/scanner/src/main/kotlin/scanner/OrtServerScanResultStorage.kt

```diff
@@ -65,15 +65,21 @@ import org.slf4j.LoggerFactory
 private val logger = LoggerFactory.getLogger(OrtServerScanResultStorage::class.java)
 
 /**
- * A class providing access to scan results.
+ * A special [ProvenanceBasedScanStorage] implementation to integrate the ORT scanner with ORT Server.
  *
- * [ScanResult] are searched by details of a [KnownProvenance] which it is associated with. If the [KnownProvenance] is
- * an [ArtifactProvenance], the URL and the hash value must match. If the [KnownProvenance] is a [RepositoryProvenance],
- * the VCS type, URL and the resolved revision must match.
+ * The [ScannerWorker] creates an instance of this class and passes it to the ORT scanner. Via this instance, the
+ * scanner can then read already existing scan results from the ORT Server database and write new scan results to it.
+ * Read and written scan results are associated to the scanner run with the provided [scannerRunId]; this establishes
+ * the link to the current ORT run.
  *
- * Read and written scan results are associated to the scanner run with the provided [scannerRunId].
+ * [ScanResult]s are searched by details of a [KnownProvenance] which it is associated with. If the [KnownProvenance]
+ * is an [ArtifactProvenance], the URL and the hash value must match. If the [KnownProvenance] is a
+ * [RepositoryProvenance], the VCS type, URL and the resolved revision must match. Note that for read operations,
+ * only the basic properties of scan results are relevant. Based on this information, the ORT scanner decides for
```

Comment:
Which are the basic properties which are relevant ?

Does this include also the issues ?

Reply:
I left this intentionally a bit vague. Currently the issues are still loaded, but I hope that a further optimization could prevent this.